api/api/models_api.go

```diff
@@ -99,3 +103,119 @@ func (c *ModelsController) GetModel(r *http.Request, vars map[string]string, bod
 
 	return Ok(model)
 }
+
+// DeleteModel delete a model given a project and model ID
+func (c *ModelsController) DeleteModel(r *http.Request, vars map[string]string, body interface{}) *Response {
+	ctx := r.Context()
+
+	projectID, err := models.ParseID(vars["project_id"])
+	if err != nil {
+		return BadRequest("Unable to parse project_id")
+	}
+	modelID, err := models.ParseID(vars["model_id"])
+	if err != nil {
+		return BadRequest("Unable to parse model_id")
+	}
+
+	_, err = c.ProjectsService.GetByID(ctx, int32(projectID))
+	if err != nil {
+		return NotFound(err.Error())
+	}
+
+	model, err := c.ModelsService.FindByID(ctx, modelID)
+	if err != nil {
+		if gorm.IsRecordNotFoundError(err) {
+			return NotFound(fmt.Sprintf("Model is not found: %s", err.Error()))
+		}
+		return InternalServerError(err.Error())
+	}
+
+	// get all model version for this model
+	versions, _, err := c.VersionsService.ListVersions(ctx, modelID, c.MonitoringConfig, service.VersionQuery{})
+	if err != nil {
+		return InternalServerError(err.Error())
+	}
+
+	var inactiveJobsInModel [][]*models.PredictionJob
+	var inactiveEndpointsInModel [][]*models.VersionEndpoint
+
+	// iterate for every version, check the active prediction job / active endpoint
+	for _, version := range versions {
+		if model.Type == "pyfunc_v2" {
+			// check active prediction job
+			// if there are any active prediction job using the model version, deletion of the model version are prohibited
+			jobs, response := c.VersionsController.getInactivePredictionJobsForDeletion(ctx, model, version)
+			if response != nil {
+				return response
+			}
+
+			// save all prediction jobs for deletion process later on
+			inactiveJobsInModel = append(inactiveJobsInModel, jobs)
+
+		}
+
+		// check active endpoints for all model type
+		// if there are any active endpoint using the model version, deletion of the model version are prohibited
+		endpoints, response := c.VersionsController.getInactiveEndpointsForDeletion(ctx, model, version)
+		if response != nil {
+			return response
+		}
+
+		// save all endpoint for deletion process later on
+		inactiveEndpointsInModel = append(inactiveEndpointsInModel, endpoints)
+
+	}
+
+	// DELETING ALL THE RELATED ENTITY
+	for index, version := range versions {
+		if model.Type == "pyfunc_v2" {
+			// delete inactive prediction jobs for model with type pyfunc_v2
+			response := c.VersionsController.deletePredictionJobs(ctx, inactiveJobsInModel[index], model, version)
+			if response != nil {
+				return response
+			}
+		}
+
+		// delete inactive endpoints for all model type
+		response := c.VersionsController.deleteVersionEndpoints(inactiveEndpointsInModel[index], version)
+		if response != nil {
+			return response
+		}
+
+		// DELETE VERSION
+		err = c.VersionsService.Delete(version)
+		if err != nil {
+			return InternalServerError(fmt.Sprintf("Delete Failed: %s", err.Error()))
```

Comment:
Nit: Delete version %d failed. It's useful to indicate what resource's deletion failed since we are trying to delete multiple versions here.
```golang
return InternalServerError(fmt.Sprintf("Delete version %d failed: %s", version.ID, err.Error()))
```